src/main/java/com/commercetools/sync/commons/utils/QueryAll.java

```diff
@@ -148,32 +157,8 @@ private void mapOrConsume(@Nonnull final List<T> pageElements) {
         if (pageElements.size() == pageSize) {
             final String lastElementId = pageElements.get(pageElements.size() - 1).getId();
             final QueryPredicate<T> queryPredicate = QueryPredicate.of(format("id > \"%s\"", lastElementId));
-            return queryPage(queryPredicate);
+            return client.execute(baseQuery.plusPredicates(queryPredicate));
```

Comment:
If you have multiple pages, will it keep appending the query?

Reply:
nope, because `plusPredicates` does not mutate the  `baseQuery` but copies and returns new instance.